src/Metadata/FluentConfiguration/EntityTypeBuilder.cs

```diff
@@ -1,99 +1,119 @@
+            return this;
         }
 
-        /// <summary>
-        /// Returns an object that can be used to configure a property of the entity type.
-        /// </summary>
-        /// <typeparam name="TProperty">Propery type.</typeparam>
-        /// <param name="propertyExpression">A lambda expression representing the property to be configured.</param>
-        /// <returns>An object that can be used to configure the property.</returns>
-        public virtual PropertyBuilder<TEntity> Property<TProperty>(
+        IEntityTypeConvertersBuilder<TEntity> IEntityTypeConvertersBuilder<TEntity>.UseConverterForProperties<TProperty, TConverter>()
+            => ((IEntityTypeConvertersBuilder<TEntity>)this).UseConverterForProperties<TProperty>(new TConverter());
+
+        IPropertyBuilder<TEntity> IEntityTypePropertyBuilder<TEntity>.Property<TProperty>(
             Expression<Func<TEntity, TProperty>> propertyExpression)
         {
+            Check.NotNull(propertyExpression, nameof(propertyExpression));
             string propertyName = PropertyName<TEntity>.GetPropertyName(propertyExpression);
-
-            if (!_propertyBuilders.TryGetValue(propertyName, out PropertyBuilder<TEntity> propertyBuilder))
+            if (_propertyBuilders.ContainsKey(propertyName))
             {
-                propertyBuilder = new PropertyBuilder<TEntity>(this, propertyName);
-                _propertyBuilders.Add(propertyName, propertyBuilder);
+                ThrowHelper.PropertyAlreadyConfigured(propertyName);
             }
 
+            var propertyBuilder = new PropertyBuilder<TEntity>(this, propertyName);
+            _propertyBuilders.Add(propertyName, propertyBuilder);
             return propertyBuilder;
         }
 
-        internal InjectionConfiguration<TEntity> Injector => _injector.Value;
-
         internal override void Build(IModelMapperInternal modelMapper)
         {
             if (!_tableName.IsNullOrWhiteSpace())
             {
                 modelMapper.SetTableName<TEntity>(_tableName);
             }
-            if (_primaryKeyBuilder != null)
+            if (!_primaryKeyPropertyName.IsNullOrWhiteSpace())
             {
-                _primaryKeyBuilder.Build(modelMapper);
+                modelMapper.SetPrimaryKey<TEntity>(_primaryKeyPropertyName, _autoIncrementType);
             }
-            if (_injector.IsValueCreated)
+            foreach (KeyValuePair<Type, IConverter> item in _propertyConverters)
             {
-                modelMapper.SetInjector<TEntity>(_injector.Value);
+                modelMapper.SetConverterForProperties<TEntity>(item.Key, item.Value);
             }
 
+            var injectionConfig = new Lazy<InjectionConfiguration<TEntity>>(() => new InjectionConfiguration<TEntity>());
             foreach (PropertyBuilder<TEntity> propertyBuilder in _propertyBuilders.Values)
             {
-                propertyBuilder.Build(modelMapper);
+                if (propertyBuilder.IsMapped)
```

Comment:
Why did you remove the Build method and replace it with several internal properties?